### DISCUSSION TITLE
Override default oauth validators in v5 typebox schema

### DISCUSSION THREAD

**Discussion by ericuldall** [feathersjs/feathers#3340] [Q&A] [answered]
I have a custom redirect flow / options and I think it's causing this error when my api starts:
```
error: Unhandled Rejection ValidationError: validation failed
// paths redacted    
{
  errors: [
    {
      instancePath: '/authentication/oauth/redirect',
      schemaPath: '#/properties/authentication/properties/oauth/properties/redirect/type',
      keyword: 'type',
      params: [Object],
      message: 'must be string'
    }
  ],
  validation: true,
  ajv: true
}
```

For redirect in my config I have the following:
```
redirect: {
  success: '/success-path',
  error: '/error-path'
}
```

How can I extend the default validation to account for this?

**Comment by howlettt** [comment, marked as answer]
I suspect you can modify the configuration.ts file to achieve this: defaultAppConfiguration -> authentication -> oauth -> redirect

https://github.com/feathersjs/feathers/blob/9e0aadcee4f6ad45d4b364b7fa6f6caf6c7ed419/packages/typebox/src/default-schemas.ts#L51

You should be able to provide your own custom config instead of using defaultAppConfiguration

**Reply by ericuldall** [reply]
Yes, I'll try that with `Type.Union` perhaps and see if it goes.

**Reply by ericuldall** [reply]
This seems to do the trick, no more error on startup:
```
export const configurationSchema = Type.Intersect([
  Type.Intersect([
    defaultAppConfiguration,
    Type.Object({
      authentication: Type.Object({
        oauth: Type.Object({
          redirect: Type.Object({
            success: Type.String(),
            error: Type.String()
          })
        })
      })
    })
  ]),
  Type.Object({
    host: Type.String(),
    port: Type.Number(),
    public: Type.String()
  })
])
```

For some reason my access token isn't getting passed through from google oauth redirect so I need to debug that to make sure it's not introduced by this change.

**Reply by ericuldall** [reply]
Missing token is unrelated the fix works!
